Packs/EWS/Integrations/EWSO365/EWSO365.py

```diff
@@ -2215,26 +2194,38 @@ def fetch_emails_as_incidents(client: EWSClient, last_run):
             client.folder_name,
             last_run.get(LAST_RUN_TIME),
             excluded_ids,
+            incidentFilter,
         )
 
         incidents = []
         incident: Dict[str, str] = {}
         demisto.debug(f'{APP_NAME} - Started fetch with {len(last_emails)} at {last_run.get(LAST_RUN_TIME)}')
         current_fetch_ids = set()
+
+        last_fetch_time = last_run.get(LAST_RUN_TIME)
+
+        last_incident_run_time = last_fetch_time
+        if isinstance(last_incident_run_time, EWSDateTime):
+            last_incident_run_time = last_incident_run_time.ewsformat()
         for item in last_emails:
             if item.message_id:
                 current_fetch_ids.add(item.message_id)
                 incident = parse_incident_from_item(item)
                 incidents.append(incident)
+                if incidentFilter == 'modified-time':
+                    item_modified_time = item.last_modified_time.ewsformat()
+                    if last_incident_run_time is None or last_incident_run_time < item_modified_time:
+                        last_incident_run_time = item_modified_time
 
                 if len(incidents) >= client.max_fetch:
                     break
 
         demisto.debug(f'{APP_NAME} - ending fetch - got {len(incidents)} incidents.')
 
-        last_fetch_time = last_run.get(LAST_RUN_TIME)
-
-        last_incident_run_time = incident.get("occurred", last_fetch_time)
+        if incidentFilter == 'modified-time':
+            pass  # Just want to make sure you know this isn't doing anything
+        else:  # default case - using 'received' time
+            last_incident_run_time = incident.get("occurred", last_fetch_time)
```

Comment:
This change is needed since changing the last_incident_run_time to last_modification_time is a bit more clear to differ the 2 flows.

```suggestion
        if incidentFilter == 'modified-time':
            last_incident_run_time = last_modification_time
        else:  # default case - using 'received' time
            last_incident_run_time = incident.get("occurred", last_fetch_time)
```